File: assets/cs2j/showcase/code/fizzbuzz/java/FizzBuzz/Core/AggregatedTransformer.java
//

package FizzBuzz.Core;

import CS2JNet.System.Collections.LCC.CSList;
import CS2JNet.System.Collections.LCC.ICollection;
import FizzBuzz.Core.ITransformer;

//   Copyright 2007 Michael Minutillo (http://wolfbyte-net.blogspot.com)
//
//   Licensed under the Apache License, Version 2.0 (the "License");
//   you may not use this file except in compliance with the License.
//   You may obtain a copy of the License at
//
//       http://www.apache.org/licenses/LICENSE-2.0
//
//   Unless required by applicable law or agreed to in writing, software
//   distributed under the License is distributed on an "AS IS" BASIS,
//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//   See the License for the specific language governing permissions and
//   limitations under the License.
/**
* Abstract base class for containers of Transforms
*/
public abstract class AggregatedTransformer   implements ITransformer
{
    private ITransformer[] transforms;
    protected ICollection<ITransformer> getTransforms() throws Exception {
        return new CSList<ITransformer>(transforms);
    }

    /**
    * Initializes a new instance of the 
    *  {@link #AggregatedTransformer}
    *  class.
    * 
    *  @param children The child transforms.
    */
    protected AggregatedTransformer(ITransformer... children) throws Exception {
        if (children == null)
            children = new ITransformer[]{  };
         
        transforms = children;
    }

    /**
    * Transforms the specified number into a string.
    * 
    *  @param number The number to transform.
    *  @return A String repesentation of the number.
    */
    public abstract String transform(int number) throws Exception ;

}

File: assets/cs2j/showcase/code/fizzbuzz/csharp/FizzBuzz.Core/Transformers/AggregatedTransformer.cs
//   Copyright 2007 Michael Minutillo (http://wolfbyte-net.blogspot.com)
//
//   Licensed under the Apache License, Version 2.0 (the "License");
//   you may not use this file except in compliance with the License.
//   You may obtain a copy of the License at
//
//       http://www.apache.org/licenses/LICENSE-2.0
//
//   Unless required by applicable law or agreed to in writing, software
//   distributed under the License is distributed on an "AS IS" BASIS,
//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//   See the License for the specific language governing permissions and
//   limitations under the License.

using System;
using System.Collections.Generic;

namespace FizzBuzz.Core
{
    /// <summary>
    /// Abstract base class for containers of Transforms
    /// </summary>
    public abstract class AggregatedTransformer : ITransformer
    {
        private readonly ITransformer[] transforms;

        protected ICollection<ITransformer> Transforms
        {
            get
            {
                return new List<ITransformer>(transforms);
            }
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="AggregatedTransformer"/> class.
        /// </summary>
        /// <param name="children">The child transforms.</param>
        protected AggregatedTransformer(params ITransformer[] children)
        {
            if (children == null)
                children = new ITransformer[] { };
            
            transforms = children;
        }

        #region ITransform Members

        /// <summary>
        /// Transforms the specified number into a string.
        /// </summary>
        /// <param name="number">The number to transform.</param>
        /// <returns>A String repesentation of the number.</returns>
        public abstract string Transform(int number);

        #endregion
    }
}
